spec/generators/dashboard_generator_spec.rb

```diff
@@ -244,18 +273,51 @@ class Person < ActiveRecord::Base
           ActiveRecord::Schema.define do
             create_table(:comments) { |t| t.references :post }
           end
+
           class Comment < ActiveRecord::Base
             belongs_to :post
           end
 
+          class Post; end
+
           run_generator ["comment"]
           load file("app/dashboards/comment_dashboard.rb")
           attrs = CommentDashboard::ATTRIBUTE_TYPES
 
-          expect(attrs[:post]).to eq(Administrate::Field::BelongsTo)
-          expect(attrs.keys).not_to include(:post_id)
+          expect(attrs["post"]).to eq(Administrate::Field::BelongsTo)
+          expect(attrs.keys).not_to include("post_id")
         ensure
-          remove_constants :Comment, :CommentDashboard
+          remove_constants :Post, :Comment, :CommentDashboard
+        end
+      end
+
+      it "detects belongs_to relationships with namespace" do
+        begin
+          ActiveRecord::Schema.define do
+            create_table(:blog_comments) { |t| t.references :post }
+          end
+
+          module Blog
+            def self.table_name_prefix
+              "blog_"
+            end
+          end
+
+          class Blog::Post; end
+
+          class Blog::Comment < ActiveRecord::Base
+            belongs_to :post
+          end
+
+          run_generator ["blog/comment"]
+          load file("app/dashboards/blog/comment_dashboard.rb")
+          attrs = Blog::CommentDashboard::ATTRIBUTE_TYPES
+
+          expect(attrs["blog/post"]).to eq(Administrate::Field::BelongsTo)
+          expect(attrs.keys).not_to include("post_id")
+          expect(attrs.keys).not_to include("blog_post_id")
+        ensure
+          remove_constants "Blog::Comment", "Blog::CommentDashboard", "Blog::Post", :Blog
```

Comment:
It will conflict with `class Blog::Post < ActiveRecord::Base` on (99 line the same file https://github.com/thoughtbot/administrate/pull/688/files#diff-9b1cf865dea4e70444fb3ffed6cecf8bR99)

I think it is a good approach to clean up state after each test, it might prevent weird test errors and false results.